flusher.go

```diff
@@ -215,25 +215,33 @@ func (s *Server) reportGlobalMetricsFlushCounts(ms metricsSummary) {
 }
 
 func (s *Server) flushS3(finalMetrics []DDMetric) {
-	start := time.Now()
+	const delimiter = '\t'
 
-	var data bytes.Buffer
-	err := json.NewEncoder(&data).Encode(finalMetrics)
+	start := time.Now()
+	csv, err := encodeDDMetricsCSV(finalMetrics, delimiter)
 	if err != nil {
 		s.logger.WithFields(logrus.Fields{
 			"metrics": len(finalMetrics),
-		}).Error("Could not marshal finalMetrics before posting to s3")
+		}).Error("Could not marshal finalMetrics before posting to s3: ", err)
 		return
 	}
 
 	// this feels dirty but oh well
-	seekableData := bytes.NewReader(data.Bytes())
+	// the AWS sdk requires seekable input
+	bts, err := ioutil.ReadAll(csv)
```

Comment:
instead of doing the ReadAll here, can we have `encodeDDMetricsCSV` return a ReadSeeker instead? that avoids the cost of reading the buffer again. we can move the comment there as well to explain why we're doing it that way